### PR TITLE
Rationalize Duration properties/methods

### LINKED ISSUE #402: Duration.Ticks is inconsistent with other properties
Currently Duration.Ticks attempts to be like the same property in 1.x. We should embrace our new nanosecond precision and make the Ticks property act more like the other properties (or rename it). We probably want a property or method which _does_ act like the current Ticks property though.

### PR DESCRIPTION
**Pull request by jskeet**
- Ticks => BclCompatibleTicks
- ToDoubleNanoseconds() => TotalNanoseconds
- Add TotalMilliseconds
- Fix typos in XML comments

Fixes #402.

### CODE REVIEW

File: src/NodaTime/Duration.cs [hunk 1]
```diff
@@ -295,26 +299,62 @@ public long Ticks
...
-        /// <value>The total number of minutes in this duration.</value>
+        /// </remarks>
+        /// <value>The total number of seconds in this duration.</value>
         public double TotalSeconds => days * (double) SecondsPerDay + nanoOfDay / (double) NanosecondsPerSecond;
 
-        // TODO: TotalMilliseconds, TotalTicks? What about TotalNanoseconds instead of ToDoubleNanoseconds?
+        /// <summary>
+        /// Gets the total number of milliseconds in this duration, as a <see cref="Double"/>.
+        /// </summary>
+        /// <remarks>This property is the <c>Duration</c> equivalent of <see cref="TimeSpan.TotalMilliseconds"/>.</remarks>
```
Comment (malcolmr) [inline src/NodaTime/Duration.cs]: Move `</remarks>` later?

File: src/NodaTime/Duration.cs [hunk 2]
```diff
@@ -295,26 +299,62 @@ public long Ticks
...
+        /// Unlike <see cref="Seconds"/>, it represents the complete duration in seconds rather than
```
Comment (malcolmr) [inline src/NodaTime/Duration.cs]: I think this para (and `<value/>` below) needs updating to talk about milliseconds.

File: src/NodaTime/Duration.cs [hunk 3]
```diff
@@ -295,26 +299,62 @@ public long Ticks
...
+        /// the whole number of seconds within the minute. So for a duration
+        /// of 10 minutes, 20 seconds and 250 milliseconds, the <c>Seconds</c> property will return 20, but <c>TotalSeconds</c>
+        /// will return 620.25.
+        /// <value>The total number of seconds in this duration.</value>
+        public double TotalMilliseconds => days * (double) MillisecondsPerDay + nanoOfDay / (double) NanosecondsPerMillisecond;
+
+        /// <summary>
+        /// Gets the total number of seconds in this duration, as a <see cref="Double"/>.
```
Comment (malcolmr) [inline src/NodaTime/Duration.cs]: Likewise not seconds here either.

File: src/NodaTime/Duration.cs [hunk 4]
```diff
@@ -295,26 +299,62 @@ public long Ticks
...
+        /// </summary>
+        /// <remarks>This property is the <c>Duration</c> equivalent of <see cref="TimeSpan.Ticks"/>,
+        /// except represented as double-precision floating point number instead of a 64-bit integer. This
+        /// is because <see cref="Duration"/> has a precision of nanoseconds, and also because the range
+        /// of 64-bit integers doesn't cover the number of possible ticks in a <see cref="Duration"/>. (The
+        /// latter is only an issue in durations outside the range of <see cref="TimeSpan"/> - in other words,
+        /// with magnitudes of over 29,000 years.)
+        /// .</remarks>
```
Comment (malcolmr) [inline src/NodaTime/Duration.cs]: nit: extraneous '.'.

File: src/NodaTime/Duration.cs [hunk 5]
```diff
@@ -254,7 +258,7 @@ internal Duration(int days, [Trusted] long nanoOfDay)
         /// </remarks>
         /// <exception cref="OverflowException">The number of ticks cannot be represented a signed 64-bit integer.</exception>
         /// <value>The total number of ticks in the duration.</value>
```
Comment (malcolmr) [inline src/NodaTime/Duration.cs]: Add a `<seealso/>` to `TotalTicks`?

File: src/NodaTime/Duration.cs [hunk 6]
```diff
@@ -295,26 +299,62 @@ public long Ticks
...
+        /// </summary>
+        /// <remarks>This property is the <c>Duration</c> equivalent of <see cref="TimeSpan.Ticks"/>,
+        /// except represented as double-precision floating point number instead of a 64-bit integer. This
+        /// is because <see cref="Duration"/> has a precision of nanoseconds, and also because the range
+        /// of 64-bit integers doesn't cover the number of possible ticks in a <see cref="Duration"/>. (The
+        /// latter is only an issue in durations outside the range of <see cref="TimeSpan"/> - in other words,
+        /// with magnitudes of over 29,000 years.)
+        /// .</remarks>
+        /// <value>The total number of ticks in this duration.</value>
+        public double TotalTicks => days * (double) TicksPerDay + nanoOfDay / (double) NanosecondsPerTick;
```
Comment (malcolmr) [inline src/NodaTime/Duration.cs]: Add a `<seealso/>` to `BclCompatibleTicks`?

File: src/NodaTime/Duration.cs (hunk 5 above)
Comment (jskeet) [inline src/NodaTime/Duration.cs]: Done.

File: src/NodaTime/Duration.cs (hunk 3 above)
Comment (jskeet) [inline src/NodaTime/Duration.cs]: Done.

File: src/NodaTime/Duration.cs (hunk 2 above)
Comment (jskeet) [inline src/NodaTime/Duration.cs]: Done.

File: src/NodaTime/Duration.cs (hunk 6 above)
Comment (jskeet) [inline src/NodaTime/Duration.cs]: Done.

**Comment by jskeet** [comment]
Thanks - squashing and merging now.

**Pull request merged**